pkg/api/validate/dynamic/dynamic.go

```diff
@@ -436,7 +436,14 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 		if oc.Properties.ProvisioningState == api.ProvisioningStateCreating {
 			if ss.SubnetPropertiesFormat != nil &&
 				ss.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
-				return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, s.Path, "The provided subnet '%s' is invalid: must not have a network security group attached.", s.ID)
+				expectedNsgID, err := subnet.NetworkSecurityGroupID(oc, s.ID)
```

Comment:
I think we are adressing symptom, not the root cause: we need to understand why in some cases (like [what we discussed](https://redhat-internal.slack.com/archives/C02ULBRS68M/p1675371441574689) in Slack recently) validation fails.

This validation should happen before any provisioning of NSG.

Reply:
The issue arises if the cluster document is picked up again and the install process rerun. This can happen if the RP panics, RP is upgraded, AKS is upgraded, or if another instance picks up a lease for an install operation.

You can emulate the failure by stopping the install process manually after [the attachNSG step](https://github.com/Azure/ARO-RP/blob/78acd1eed34b9c63af50ed8a2485a61836429cfa/pkg/cluster/install.go#L246) has been run.

The RP should be able to rerun the install process if it fails due to the reasons listed above, but currently it cannot.

Reply:
Yeah the cluster install should always be idempotent, but it's not in this specific instance.  (unless this PR is merged)

Reply:
@m1kola I don't think there is anything else to add to this PR, since its intention is to unblock re-queued installs, would you mind giving it another look-over?

Reply:
Thanks for providing more context. Makes sense.